package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -58,30 +58,6 @@ dhcp_calc() {
 	echo "$res"
 }
 
-dhcp_check() {
-	local ifname="$1"
-	local stamp="${BASEDHCPSTAMPFILE_CFG}.${ifname}.dhcp"
-	local rv=0
-
-	[ -s "$stamp" ] && return $(cat "$stamp")
-
-	# If there's no carrier yet, skip this interface.
-	# The init script will be called again once the link is up
-	case "$(devstatus "$ifname" | jsonfilter -e @.carrier)" in
-		false) return 1;;
-	esac
-
-	udhcpc -n -q -s /bin/true -t 1 -i "$ifname" >&- && rv=1 || rv=0
-
-	[ $rv -eq 1 ] && \
-		logger -t dnsmasq \
-			"found already running DHCP-server on interface '$ifname'" \
-			"refusing to start, use 'option force 1' to override"
-
-	echo $rv > "$stamp"
-	return $rv
-}
-
```

Comment:
All times I did one of such mistakes (leaving DHCP on in LAN when connecting it to a lan where there was already a DHCP server of a device with stock firmware or dd-wrt) the openwrt/lede device's DHCP was still active, and I experienced the expected breakage for that setup. 
I never thought that there was a check about that, are you sure the current check works at all?
Or what I experienced is unrelated?